cardinal/cardinal_test.go

```diff
@@ -59,9 +59,11 @@ func TestShutdownViaSignal(t *testing.T) {
 		_, err := cardinal.CreateMany(worldCtx, wantNumOfEntities, Foo{})
 		assert.NilError(t, err)
 	})
+	otherFinish := make(chan bool)
```

Comment:
optional: You could use a [WaitGroup](https://pkg.go.dev/sync#WaitGroup) here.

Reply:
agree there - we should remove both channels and use waitgroup for their replacements